src/Chullo.php

```diff
@@ -58,56 +62,49 @@ public function getBaseUri() {
      * @param array     $headers        HTTP Headers
      * @param string    $transaction    Transaction id
      *
-     * @return mixed    String or binary content if 200. Null if 304.
+     * @return mixed    Full response if found.  Null otherwise.
      */
-    public function getResource($uri = "",
-                                $headers = [],
-                                $transaction = "") {
-        // Set headers
-        $options = ['headers' => $headers];
-
-        // Ensure uri takes transaction into account.
-        $uri = $this->prepareUri($uri, $transaction);
-
-        // Send the request.
-        $response = $this->client->request(
-            'GET',
+    public function getResource(
+        $uri = "",
+        $headers = [],
+        $transaction = ""
+    ) {
+        $response = $this->api->getResource(
             $uri,
-            $options
+            $headers,
+            $transaction
         );
-
-        $code = $response->getStatusCode();
-
-        if ($code == 304) {
+        if ($response->getStatusCode() != 200) {
             return null;
         }
 
-        return $response->getBody();
+        return $response;
```

Comment:
Hi, i feel having the whole response is better. e.g, One need is to be able to now what mime type i'm getting back when getting a binary resource.